lib/kb_Metrics/metricsdb_controller.py

```diff
@@ -515,44 +512,43 @@ def join_app_task_ujs(self, exec_tasks, exec_apps, ujs_jobs):
             u_j_s['creation_time'] = u_j_s.pop('created')
             u_j_s['modification_time'] = u_j_s.pop('updated')
             u_j_s['estcompl'] = j.get('estcompl', None)
-            u_j_s['time_info'] = [u_j_s.get('creation_time'),
-                                  u_j_s.get('modification_time'),
-                                  u_j_s['estcompl']]
 
-            if u_j_s.get('authstrat', None) == 'kbaseworkspace':
-                u_j_s['wsid'] = u_j_s.get('authparam')
+            authparam = u_j_s.pop('authparam')
+            authstrat = u_j_s.pop('authstrat')
+            if authstrat == 'kbaseworkspace':
+                u_j_s['wsid'] = authparam
 
             if u_j_s.get('desc'):
-                desc = u_j_s['desc'].split()[-1]
+                desc = u_j_s.pop('desc').split()[-1]
                 if '.' in desc:
                     u_j_s['method'] = desc
 
             # Assuming complete, error and status all exist in the records returned
-            if not j.get('error'):
-                if j.get('complete'):
+            if not u_j_s.get('error'):
+                if u_j_s.get('complete'):
                     u_j_s['job_state'] = 'completed'
-                elif j.get('status') in ["Initializing", 'queued']:
-                    u_j_s['job_state'] = j['status']
-                elif j.get('status') in ['canceled', 'cancelled']:
+                elif u_j_s.get('status') in ["Initializing", 'queued']:
+                    u_j_s['job_state'] = u_j_s['status']
+                elif u_j_s.get('status') in ['canceled', 'cancelled']:
                     u_j_s['job_state'] = 'canceled'
-                elif j.get('started'):
+                elif u_j_s.get('exec_start_time'):
                     u_j_s['job_state'] = 'in-progress'
-                elif j['created'] == j['updated']:
+                elif u_j_s['creation_time'] == u_j_s['modification_time']:
                     u_j_s['job_state'] = 'not-started'
-                elif j['created'] < j['updated'] and not j.get('started'):
+                elif (u_j_s['creation_time'] < u_j_s['modification_time'] and
+                        not u_j_s.get('exec_start_time')):
                     u_j_s['job_state'] = 'queued'
                 else:
                     u_j_s['job_state'] = 'unknown'
             else:
-                u_j_s['job_state'] = 'suspended'
+                u_j_s['job_state'] = 'suspend'
```

Comment:
keep it as 'suspend'?

Reply:
Yes...after I checked the exec_engine and found it was called 'suspend' there.  To keep it straightforward.   As you have seen that 'canceled' and 'cancelled' also occur, that one I consolidated.

Reply:
why? just change it to 'suspended'? At least they are all adjectives.

Reply:
That's a no-brainer.  But if the source db use 'suspend' I prefer to stay with it.

Reply:
okay. So you want to fix 'cancelled' but not 'suspended'.